src/mandelbrot.rs

```diff
@@ -0,0 +1,104 @@
+// The Computer Language Benchmarks Game
+// http://benchmarksgame.alioth.debian.org/
+//
+// contributed by Matt Watson
+// contributed by TeXitoi
+
+use std::io::Write;
+use std::thread;
+const THREADS: usize = 20;
+const MAX_ITER: usize = 50;
+const VLEN: usize = 8;
+const ZEROS: Vecf64 = [0.0; VLEN];
+
+pub type Vecf64 = [f64; VLEN];
+
+fn mul2 (x: Vecf64, y: Vecf64) -> Vecf64 {
+    let mut res = ZEROS;
+    for i in 0..VLEN {res[i] = x[i] * y[i];}
+    res
+}
+fn add2 (x: Vecf64, y: Vecf64) -> Vecf64 {
+    let mut res = ZEROS;
+    for i in 0..VLEN {res[i] = x[i] + y[i];}
+    res
+}
+fn sub2 (x: Vecf64, y: Vecf64) -> Vecf64 {
+    let mut res = ZEROS;
+    for i in 0..VLEN { res[i] = x[i] - y[i]; }
+    res
+}
+
+pub fn mbrot8(cr: Vecf64, ci: Vecf64) -> u8 {
+    let mut zr = cr;
+    let mut zi = ci;
+    let mut esc_bits = 0;
+    for _ in 0..MAX_ITER {
+        // Find Re(z)^2 and Im(z)^2
+        let rr  = mul2(zr,zr);
+        let ii  = mul2(zi,zi);
+        // Check if we escape
+        // May as well store this info in
+        // same byte as output
+        let mag = add2(rr, ii);
+        for i in 0..VLEN {
+            if mag[i] > 4.0 {esc_bits |= 128 >> i}
+        }
+        // If no more work, break early
+        if esc_bits == 0xff {break}
+        // Find Im(z^2)
+        let ir = mul2(zr, zi);
+        // Set Re(z^2)
+        zr = sub2(rr, ii);
+        // Set Im(z^2)
+        zi = add2(ir, ir);
+        // Add c
+        zr = add2(zr, cr);
+        zi = add2(zi, ci);
+    }
+    !esc_bits
+}
+
+fn main() {
+
+    let size = std::env::args_os().nth(1)
+        .and_then(|s| s.into_string().ok())
+        .and_then(|n| n.parse().ok())
+        .unwrap_or(200);
+    let inv = 2.0 / size as f64;
+    let mut xvals = vec![0.0; size];
+    let mut yvals = vec![0.0; size];
+    for i in 0..size {
+        xvals[i] = i as f64 * inv - 1.5;
+        yvals[i] = i as f64 * inv - 1.0;
+    }
+    let xloc = &xvals;
+    let yloc = &yvals;
+    
+    let handles: Vec<_> = (0..THREADS).map(|e| {
+        let xloc = xloc.to_vec();
+        let yloc = yloc.to_vec();
+        thread::spawn(move || {
+        let mut rows = vec![vec![0 as u8; size / 8]; size / THREADS];
```

Comment:
Yes, that's correct. I (or you -- I'd welcome reading your improvements in other areas as well) need to put some special case handling for the last thread.

I also had the thought that -- seeing as the cost of opening stdout seems small -- that the threads could be gathered more than once and the output streamed as we go somewhat (one could even spawn the next iteration of worker threads and stream while they are working). A side benefit would be that the memory numbers would be a small fraction of the g++\#8 version. 

I don't really grok things well enough to actually implement this though.